### PR TITLE
Clarify that "Restrict account creation to specified email domains" setting only applies to email login

### PR DESCRIPTION
**Pull request by jasonblais**
#### Summary
Clarify that "Restrict account creation to specified email domains" setting only applies to email login

**Why only have the setting apply to email login?**
Because if an admin uses an SSO (OAuth/GitLab, SAML, LDAP), they are more likely to restrict who can login to the server via their SSO provider. 

**Is "email login" appropriate terminology for help text?**
It's consistent with help text in System Console > Email Auth > Enable sign-in with username

#### Ticket Link
N/A, but a follow-up to this PR https://github.com/mattermost/mattermost-server/pull/10209

#### Checklist
- [X] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates

### CODE REVIEW

File: i18n/en.json [hunk 1]
```diff
@@ -1332,7 +1332,7 @@
   "admin.team.openServerTitle": "Enable Open Server: ",
   "admin.team.restrict_direct_message_any": "Any user on the Mattermost server",
   "admin.team.restrict_direct_message_team": "Any member of the team",
-  "admin.team.restrictDescription": "Teams and user accounts can only be created from a specific domain (e.g. \"mattermost.org\") or list of comma-separated domains (e.g. \"corp.mattermost.com, mattermost.org\").",
+  "admin.team.restrictDescription": "Teams and user accounts can only be created from a specific domain (e.g. \"mattermost.org\") or list of comma-separated domains (e.g. \"corp.mattermost.com, mattermost.org\"). This setting only affects email login.",
```
Comment (DSchalla) [inline i18n/en.json]: Should it be "Teams and user accounts" or only "User accounts"? In which case does it affect Team creation?

Comment (jasonblais) [reply i18n/en.json]: @DSchalla Good catch. This is a remnant of a (very) old system when team and user accounts could be restricted based on domain.

Updated to only say "User accounts".

**Comment by jasonblais** [comment]
@DSchalla Any further feedback or looks good?

**Comment by jasonblais** [comment]
Can you or @cpanato help with the rebase 🙏

**Comment by cpanato** [comment]
@jasonblais done

**Comment by jasonblais** [comment]
Thank you!! Ready to merge as well (no write permissions)

**Comment by DSchalla** [comment]
@jasonblais Done :-)

**Pull request merged**